plugins/src/gcc/imp.rs

```diff
@@ -0,0 +1,1270 @@
+            .unwrap()
+            .arrival;
+
+        while last_arrival - self.oldest_packet_in_window_ts() > *PACKETS_RECEIVED_WINDOW {
+            self.last_received_packets.remove(0);
+        }
+    }
+
+    /// Returns the effective received bitrate during the last PACKETS_RECEIVED_WINDOW
+    fn effective_bitrate(&self) -> Bitrate {
+        let len = self.last_received_packets.len();
+        if len == 0 {
+            return 0;
+        }
+
+        let duration = self.last_received_packets.get(len - 1).unwrap().arrival
+            - self.last_received_packets.get(0).unwrap().arrival;
+        let bits = self
+            .last_received_packets
+            .iter()
+            .map(|p| p.size as f64)
+            .sum::<f64>()
+            * 8.;
+
+        (bits
+            / (duration.num_nanoseconds().unwrap() as f64
+                / gst::ClockTime::SECOND.nseconds() as f64)) as Bitrate
+    }
+
+    fn oldest_packet_in_window_ts(&self) -> Duration {
+        self.last_received_packets.get(0).unwrap().arrival
+    }
+
+    fn update_rtts(&mut self, packets: &Vec<Packet>) {
+        let mut rtt = Duration::nanoseconds(i64::MAX);
+        let now = ts2dur(self.clock.time().unwrap());
+        for packet in packets {
+            rtt = (now - packet.departure).min(rtt);
+        }
+
+        self.rtts.push_back(rtt);
+        if self.rtts.len() > ROUND_TRIP_TIME_WINDOW_SIZE {
+            self.rtts.pop_front();
+        }
+    }
+
+    fn rtt(&self) -> Duration {
+        Duration::nanoseconds(
+            (self
+                .rtts
+                .iter()
+                .map(|d| d.num_nanoseconds().unwrap() as f64)
+                .sum::<f64>()
+                / self.rtts.len() as f64) as i64,
+        )
+    }
+
+    fn update(&mut self, packets: &Vec<Packet>) {
+        self.update_rtts(packets);
+        let mut lost_packets = 0.;
+        for pkt in packets {
+            if pkt.arrival.is_zero() {
+                lost_packets += 1.;
+                continue;
+            }
+
+            self.update_last_received_packets(*pkt);
+
+            if self.group.arrival.is_none() {
+                self.group.add(*pkt);
+
+                continue;
+            }
+
+            if pkt.arrival < self.group.arrival.unwrap() {
+                // ignore out of order arrivals
+                continue;
+            }
+
+            if pkt.departure >= self.group.departure {
+                if self.group.inter_departure_time_pkt(pkt) < *BURST_TIME {
+                    self.group.add(*pkt);
+                    continue;
+                }
+
+                if self.group.inter_arrival_time_pkt(pkt) < *BURST_TIME
+                    && self.group.inter_delay_variation_pkt(pkt) < Duration::zero()
+                {
+                    self.group.add(*pkt);
+                    continue;
+                }
+
+                let group = mem::replace(&mut self.group, PacketGroup::new());
```

Comment:
```suggestion
                let group = mem::take(&mut self.group);
```

If you implement `Default` for `PacketGroup`. Slightly nicer IMHO :)